core/src/main/java/org/frankframework/http/authentication/AbstractOauthAuthenticator.java

```diff
@@ -0,0 +1,133 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package org.frankframework.http.authentication;
+
+import org.apache.http.auth.Credentials;
+
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.EntityUtils;
+import org.apache.logging.log4j.Logger;
+
+import org.frankframework.http.AbstractHttpSession;
+import org.frankframework.task.TimeoutGuard;
+import org.frankframework.util.DateFormatUtils;
+import org.frankframework.util.JacksonUtils;
+import org.frankframework.util.LogUtil;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+import static org.frankframework.util.StreamUtil.DEFAULT_INPUT_STREAM_ENCODING;
+
+public abstract class AbstractOauthAuthenticator implements IAuthenticator {
+
+	protected final Logger log = LogUtil.getLogger(this);
+
+	protected final AbstractHttpSession session;
+	protected final URI authorizationEndpoint;
+	protected final int overwriteExpiryMs;
+
+	private String accessToken;
+	private long accessTokenRefreshTime;
+
+	AbstractOauthAuthenticator(final AbstractHttpSession session) throws HttpAuthenticationException {
+		this.session = session;
+		try {
+			this.authorizationEndpoint = new URI(session.getTokenEndpoint());
+		} catch (URISyntaxException e) {
+			throw new HttpAuthenticationException(e);
+		}
+		this.overwriteExpiryMs = session.getTokenExpiry() * 1000;
+	}
+
+	protected abstract HttpEntityEnclosingRequestBase createRequest(Credentials credentials) throws HttpAuthenticationException;
+
+	protected HttpEntityEnclosingRequestBase createPostRequestWithForm(URI uri, List<BasicHeader> formParameters) throws HttpAuthenticationException {
+		try {
+			UrlEncodedFormEntity body = new UrlEncodedFormEntity(formParameters, DEFAULT_INPUT_STREAM_ENCODING);
+
+			HttpPost request = new HttpPost(uri);
+			request.addHeader(body.getContentType());
```

Comment:
Deze call zou niet moeten hoeven dacht ik, de contenttype van de entity is de default? In de HttpSender doen we het iig niet 😃

Reply:
De `Content-Type` header wordt niet toegevoegd aan de request als ik die regel weghaal.

Reply:
Dat zou betekenen dat het in de rest can de (huidige) code fout zou zijn. Overal wordt het content type aan de entity gehangen, en de `HttpEntityEnclosingRequestBase` (HttpPost) zou deze klakkeloos over moeten nemen 🍓